vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -41,12 +46,33 @@ static int signum(int x) {
         return 0;
 }
 
-typedef std::vector<std::tuple<std::pair<int, int>, std::pair<int, int>, Cost_Entry, ConnectionBoxId>> t_routing_cost_map;
-static void run_dijkstra(int start_node_ind,
-                         t_routing_cost_map* routing_cost_map);
+/* Tuple key is: segment, connection_box, dx, dy */
+typedef std::tuple<int, ConnectionBoxId, int, int> routing_data_key_t;
```

Comment:
std::tuple has `<` and `==` defined, which is nice if transitioning between hashed and ordered data.  If std::tuple had std::hash defined, it would even better.  I believe in the future C++ standard this gets fixed.

Reply:
I've started trying to convert from tuple to struct, and the std::unordered_map requires `operator ==`, so ya.  Anyways I've converted to using a struct, running a test to check for runtime impacts.